src/ILCompiler.ReadyToRun/src/IBC/ibcdatamodel.cs

```diff
@@ -0,0 +1,537 @@
+        ModuleRefProfilingData = CONSTANT.FirstTokenFlagSection + ((int)CorTokenType.mdtModuleRef >> 24),
+        TypeSpecProfilingData = CONSTANT.FirstTokenFlagSection + ((int)CorTokenType.mdtTypeSpec >> 24),
+        ImplMapProfilingData = CONSTANT.FirstTokenFlagSection + ((int)CorTokenType.mdtImplMap >> 24),
+        FieldRVAProfilingData = CONSTANT.FirstTokenFlagSection + ((int)CorTokenType.mdtFieldRVA >> 24),
+        AssemblyProfilingData = CONSTANT.FirstTokenFlagSection + ((int)CorTokenType.mdtAssembly >> 24),
+        AssemblyRefProfilingData = CONSTANT.FirstTokenFlagSection + ((int)CorTokenType.mdtAssemblyRef >> 24),
+        FileProfilingData = CONSTANT.FirstTokenFlagSection + ((int)CorTokenType.mdtFile >> 24),
+        ExportedTypeProfilingData = CONSTANT.FirstTokenFlagSection + ((int)CorTokenType.mdtExportedType >> 24),
+        ManifestResourceProfilingData = CONSTANT.FirstTokenFlagSection + ((int)CorTokenType.mdtManifestResource >> 24),
+        NestedClassProfilingData = CONSTANT.FirstTokenFlagSection + ((int)CorTokenType.mdtNestedClass >> 24),
+        GenericParamProfilingData = CONSTANT.FirstTokenFlagSection + ((int)CorTokenType.mdtGenericParam >> 24),
+        MethodSpecProfilingData = CONSTANT.FirstTokenFlagSection + ((int)CorTokenType.mdtMethodSpec >> 24),
+        GenericParamConstraintProfilingData = CONSTANT.FirstTokenFlagSection + ((int)CorTokenType.mdtGenericParamConstraint >> 24),
+
+        StringPoolProfilingData,
+        GuidPoolProfilingData,
+        BlobPoolProfilingData,
+        UserStringPoolProfilingData,
+
+        GenericTypeProfilingData = 63,    // 0x3f
+        SectionFormatCount = 64,    // 0x40
+
+        SectionFormatInvalid = -1,
+    }
+
+    public enum BlobType : int
+    {
+        // IMPORTANT: Keep the first four enums together in the same order and at the
+        //  very begining of this enum.  See MetaModelPub.h for the order
+
+        MetadataStringPool = 0,
+        MetadataGuidPool = 1,
+        MetadataBlobPool = 2,
+        MetadataUserStringPool = 3,
+
+        FirstMetadataPool = MetadataStringPool,
+        LastMetadataPool = MetadataUserStringPool,
+
+        ParamTypeSpec = 4,    // Instantiated Type Signature
+        ParamMethodSpec = 5,    // Instantiated Method Signature
+        ExternalNamespaceDef = 6,    // External Namespace Token Definition 
+        ExternalTypeDef = 7,    // External Type Token Definition
+        ExternalSignatureDef = 8,    // External Signature Definition
+        ExternalMethodDef = 9,    // External Method Token Definition
+
+        IllegalBlob = 10,    // Failed to allocate the blob
+
+        EndOfBlobStream = -1
+    }
+
+    public enum ModuleId : uint
+    {
+        CurrentModule = 0,    // Tokens are encoded/decoded using current modules metadata
+        ExternalModule = 1,    // Tokens are (or will be) encoded/decoded using ibcExternalType and ibcExternalMethod
+    }
+
+    public enum CorElementType : byte
```

Comment:
Do we plan on keeping this source file in sync with some external version? Otherwise I would prefer not to introduce yet another copy of the constants we already have in FixupConstants, most notably CorElementType and ReadyToRunMethodSigFlags.

Reply:
These don't need to be kept up to date. I missed the stuff in FixupConstants when I went looking as I wrote this logic.